code/game/mecha/mecha.dm

```diff
@@ -342,7 +342,10 @@
 /obj/mecha/examine(mob/user)
 	. = ..()
 	var/hide_weapon = locate(/obj/item/mecha_parts/concealed_weapon_bay) in contents
-	var/hidden_weapon = hide_weapon ? (locate(/obj/item/mecha_parts/mecha_equipment/weapon) in equipment) : null
+	var/obj/item/mecha_parts/mecha_equipment/melee_weapon/hidden_melee_weapon = locate(/obj/item/mecha_parts/mecha_equipment/melee_weapon) in equipment
+	if(hidden_melee_weapon && !(hidden_melee_weapon.mech_flags & EXOSUIT_MODULE_COMBAT))
+		hidden_melee_weapon = null
+	var/hidden_weapon = hide_weapon ? (locate(/obj/item/mecha_parts/mecha_equipment/weapon) in equipment)||hidden_melee_weapon : null
```

Comment:
are you trying to locate `/obj/item/mecha_parts/mecha_equipment/weapon` in equpment or `hidden_melee_weapon` ?

Reply:
Both, as weapon refers to mech guns, and hidden_melee_weapon refers to mech swords and other melee stuff.

Reply:
wait nvm